api/api/src/main/scala/uk/ac/wellcome/platform/api/Router.scala

```diff
@@ -222,6 +240,30 @@ class Router(elasticClient: ElasticClient,
       }
     }
 
+  def retrieveTree(index: Index,
+                   work: IdentifiedWork,
+                   expandedPaths: List[String])
+    : Future[Option[(CollectionTree, List[String])]] =
+    work.data.collection
+      .map {
+        case Collection(_, path) =>
+          val paths = path :: expandedPaths
```

Comment:
Call this `allPaths` to increase the difference from `path`? I like to keep variable names different.